frontend/src/Components/HomePage/UpcomingTrips/SeatBidding.jsx

```diff
@@ -0,0 +1,231 @@
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { io } from 'socket.io-client';
+import { API_BASE_URL } from '../../../config';
+import './SeatBidding.css';
+
+/**
+ * SeatBidding component for managing seat bidding functionality
+ * @param {Object} props - Component props
+ * @param {Object} props.trip - Trip information
+ * @param {string} props.flightId - ID of the flight
+ * @param {Function} props.onBiddingCompleted - Callback function when bidding is completed
+ */
+const SeatBidding = ({ trip, flightId, onBiddingCompleted }) => {
+    const [activeBiddings, setActiveBiddings] = useState([]);
+    const [bidAmounts, setBidAmounts] = useState({});
+    const [isLoading, setIsLoading] = useState(true);
+    const [message, setMessage] = useState('');
+
+    /**
+     * Fetch active biddings for the current flight
+     */
+    const fetchActiveBiddings = async () => {
+        try {
+            const response = await axios.get(`${API_BASE_URL}/bidding/active/${flightId}`, { withCredentials: true });
+            setActiveBiddings(response.data);
+        } catch (error) {
+            // Error handling for failed fetch
+        } finally {
+            setIsLoading(false);
+        }
+    };
+
+    useEffect(() => {
+        const newSocket = io(API_BASE_URL, { withCredentials: true });
+
+        newSocket.on('connect', () => newSocket.emit('join-flight', flightId));
+        newSocket.on('new-bid', (data) => {
+            fetchActiveBiddings();
```

Comment:
Yeah it makes sense to fetch active biddings once entering Flight Details Page, but I'd avoid re-fetching all data here. We have the information of the new bid so we should just update the state to append the new one, something like:

`setActiveBiddings([...activeBiddings, data])`

Please let me know if that doesn't work.

Reply:
fixed it. Will commit the changes in next PR!